### PR TITLE
feat: add request translator for spec to deployment + schema

### PR DESCRIPTION
**Pull request by jhamon**
## Summary

Adds `_translate_legacy_request()` method to `PineconeDBControlRequestFactory` that converts legacy spec-based index creation parameters to the new deployment + schema format required by the alpha API.

## Problem

The new Full-Text Search API uses a different structure for index creation:
- **Old format**: Uses `spec` (ServerlessSpec/PodSpec/ByocSpec) with top-level `dimension` and `metric` parameters
- **New format**: Uses `deployment` (with `deployment_type`) and `schema` (with field definitions)

To maintain backward compatibility while supporting the new API, we need translation logic that converts the old format to the new format.

## Solution

Added `_translate_legacy_request()` static method that:

1. **Converts spec types to deployment**:
   - `ServerlessSpec(cloud, region)` → `{"deployment_type": "serverless", "cloud": ..., "region": ...}`
   - `PodSpec(environment, ...)` → `{"deployment_type": "pod", "environment": ..., ...}`
   - `ByocSpec(environment)` → `{"deployment_type": "byoc", "environment": ...}`

2. **Converts vector parameters to schema**:
   - `dimension` + `metric` + `vector_type="dense"` → `schema.fields._values` (dense_vector field)
   - `vector_type="sparse"` → `schema.fields._sparse_values` (sparse_vector field)

3. **Provides sensible defaults**:
   - Defaults to "cosine" metric for dense vectors
   - Defaults to "dotproduct" metric for sparse vectors

4. **Validates inputs**:
   - Requires `dimension` for dense vector indexes
   - Raises clear errors for invalid spec types

## Usage Example

```python
from pinecone.db_control.request_factory import PineconeDBControlRequestFactory
from pinecone import ServerlessSpec

# Translate legacy request format
deployment, schema = PineconeDBControlRequestFactory._translate_legacy_request(
    spec=ServerlessSpec(cloud="aws", region="us-east-1"),
    dimension=1536,
    metric="cosine",
    vector_type="dense"
)

# Returns:
# deployment = {"deployment_type": "serverless", "cloud": "aws", "region": "us-east-1"}
# schema = {"fields": {"_values": {"type": "dense_vector", "dimension": 1536, "metric": "cosine"}}}
```

## Testing

Added comprehensive unit tests covering:
- All spec types (Serverless, Pod, BYOC)
- Both object and dict-based spec formats
- Dense and sparse vector translations
- Default metric handling
- Edge cases (missing dimension, invalid spec types)

## Related Issues

- Linear: SDK-104
- Part of the Full-Text Search SDK implementation

## Value

This translation layer enables backward compatibility by allowing existing code using the legacy `spec`-based API to work with the new schema-based API. It's a critical building block for the FTS feature set, as it allows the SDK to support both old and new index creation patterns seamlessly.

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it introduces new translation/validation logic that will shape index-creation payloads (including defaults and error conditions) and could affect backward-compatibility behavior if edge cases differ from the legacy format.
> 
> **Overview**
> Adds `PineconeDBControlRequestFactory._translate_legacy_request()` to convert legacy index creation inputs (`spec` plus `dimension`/`metric`/`vector_type`) into the new `deployment` and `schema.fields` structures, including enum-to-string normalization, sensible metric defaults (dense→`cosine`, sparse→`dotproduct`), and stricter validation (e.g., dense requires `dimension`, invalid `vector_type` errors).
> 
> Expands unit coverage for the translator across serverless/pod/BYOC (object and dict specs), defaulting behavior, zero-value preservation, and error cases; also simplifies `test_deployment.py` to import deployment models directly from `pinecone` instead of dynamically loading the module.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 6b4360f84c8418cf428b964612cf852f0b7621b7. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: pinecone/db_control/request_factory.py [hunk 1]
```diff
@@ -361,6 +367,119 @@ def __parse_index_spec(spec: Dict | ServerlessSpec | PodSpec | ByocSpec) -> Inde
...
+                    pod_type=pod_spec.get("pod_type", "p1.x1"),
+                    replicas=pod_spec.get("replicas", 1),
+                    shards=pod_spec.get("shards", 1),
+                    pods=pod_spec.get("pods"),
+                )
+                deployment_dict = deployment.to_dict()
+            elif "byoc" in spec:
+                byoc_spec = spec["byoc"]
+                deployment = ByocDeployment(environment=byoc_spec.get("environment", ""))
+                deployment_dict = deployment.to_dict()
```
[the remark this answers is not in the document]
Comment (jhamon) [reply pinecone/db_control/request_factory.py:444]: ✅ Fixed in commit aad12443. Added `convert_enum_to_string()` calls for `cloud`, `region`, `environment`, and `pod_type` when processing dict-based specs, matching the behavior of `__parse_index_spec`.

File: pinecone/db_control/request_factory.py [hunk 2]
```diff
@@ -361,6 +367,119 @@ def __parse_index_spec(spec: Dict | ServerlessSpec | PodSpec | ByocSpec) -> Inde
...
+            deployment = ServerlessDeployment(cloud=spec.cloud, region=spec.region)
+            deployment_dict = deployment.to_dict()
+        elif isinstance(spec, PodSpec):
+            # PodDeployment requires pod_type, but PodSpec defaults to "p1.x1"
+            pod_type = spec.pod_type if spec.pod_type is not None else "p1.x1"
+            deployment = PodDeployment(
+                environment=spec.environment,
+                pod_type=pod_type,
+                replicas=spec.replicas or 1,
+                shards=spec.shards or 1,
```
[the remark this answers is not in the document]
Comment (jhamon) [reply pinecone/db_control/request_factory.py]: ✅ Fixed in commit aad12443. Changed from `spec.replicas or 1` to `spec.replicas if spec.replicas is not None else 1` to preserve zero values consistently with dict-based handling.

File: pinecone/db_control/request_factory.py [hunk 3]
```diff
@@ -361,6 +367,119 @@ def __parse_index_spec(spec: Dict | ServerlessSpec | PodSpec | ByocSpec) -> Inde
...
+            # Dense vector: use _values field
+            if dimension is None:
+                raise ValueError("dimension is required for dense vector indexes")
+            if metric is None:
+                metric = Metric.COSINE.value  # Default for dense vectors
+            dense_field = DenseVectorField(dimension=dimension, metric=metric)
+            schema_dict["fields"]["_values"] = dense_field.to_dict()
+        else:
+            # No vector type specified - return empty schema fields
+            pass
```
[the remark this answers is not in the document]
Comment (jhamon) [reply pinecone/db_control/request_factory.py]: ✅ Fixed in commit aad12443. Added validation to raise `ValueError` with a clear error message when `vector_type` is invalid, instead of silently returning empty schema.

File: tests/unit/models/test_deployment.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    sys.modules[module_name] = module
+    with open(module.__file__) as f:
+        exec(compile(f.read(), module.__file__, "exec"), module.__dict__)
+    return module
+
+
+_deployment = _load_deployment_module()
+ServerlessDeployment = _deployment.ServerlessDeployment
+ByocDeployment = _deployment.ByocDeployment
+PodDeployment = _deployment.PodDeployment
```
[the remark this answers is not in the document]
Comment (jhamon) [reply tests/unit/models/test_deployment.py]: ✅ Fixed in commit aad12443. Replaced the complex `_load_deployment_module()` function with standard imports: `from pinecone import ServerlessDeployment, ByocDeployment, PodDeployment`. The classes are properly exported via `pinecone/__init__.py`, so standard imports work correctly.
